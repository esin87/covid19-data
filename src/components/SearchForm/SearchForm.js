import React from 'react';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';

const SearchForm = ({ handleSubmit, handleChange, searchString }) => {
	return (
		<Form onSubmit={handleSubmit}>
			<Form.Group>
				<h2>Country Search</h2>
				<Form.Control
					size='lg'
					placeholder='Search by country name'
					type='text'
					name='searchString'
					required
					onChange={handleChange}
					value={searchString}
				/>
				<Form.Text className='text-muted'>
					Search by full or partial country name. No abbreviations, i.e.,
					"United States", not "USA".
				</Form.Text>
			</Form.Group>
			<Button type='submit'>Submit</Button>
		</Form>
	);
};

export default SearchForm;
