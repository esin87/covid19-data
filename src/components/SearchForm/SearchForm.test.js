import React from 'react';
import SearchForm from './SearchForm';
import { mount, render } from 'enzyme';

describe('SearchForm component', () => {
	let component;
	const searchString = 'United States';
	function handleSubmit() {
		return;
	}
	function handleChange() {
		return;
	}
	// const data = {
	// 	Country: 'Switzerland',
	// 	CountryCode: 'CH',
	// 	Lat: '46.82',
	// 	Lon: '8.23',
	// 	Confirmed: 20505,
	// 	Deaths: 666,
	// 	Recovered: 6415,
	// 	Active: 13424,
	// 	Date: '2020-04-05T00:00:00Z',
	// 	LocationID: '628d4f12-6ebe-4fa9-b046-77ff0b894a4e',
	// };

	beforeEach(() => {
		component = mount(
			<SearchForm
				searchString={searchString}
				handleSubmit={handleSubmit}
				handleChange={handleChange}
			/>
		);
	});

	it('renders', () => {
		render(component);
	});

	it('should match the snapshot', () => {
		expect(component.html()).toMatchSnapshot();
	});

	it('should have an h2 with appropriate text', () => {
		expect(component.find('h2').text()).toBe('Country Search');
	});

	it('should have a muted text class with appropriate text', () => {
		expect(component.find('.text-muted').at(0).text()).toBe(
			'Search by full or partial country name. No abbreviations, i.e., "United States", not "USA".'
		);
	});

	it('should have a button with appropriate text', () => {
		expect(component.find('button').text()).toBe('Submit');
	});

	it('receives the appropriate props', () => {
		expect(component.props()).toEqual({
			searchString: 'United States',
			handleChange: expect.any(Function),
			handleSubmit: expect.any(Function),
		});
	});
});
