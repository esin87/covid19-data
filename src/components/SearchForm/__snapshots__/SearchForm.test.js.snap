// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component should match the snapshot 1`] = `"<form class=\\"\\"><div class=\\"form-group\\"><h2 style=\\"margin-bottom: 1.5rem;\\">Country Search</h2><input placeholder=\\"Search by country name\\" name=\\"searchString\\" required=\\"\\" type=\\"text\\" class=\\"form-control form-control-lg\\" value=\\"United States\\"><small class=\\"text-muted form-text\\">Search by full or partial country name. No abbreviations, i.e., \\"United States\\", not \\"USA\\".</small></div><button type=\\"submit\\" class=\\"btn btn-primary\\">Submit</button></form>"`;
